src/ui/MissingTerm/MissingTermManager.ts

```diff
@@ -80,14 +97,37 @@ export class MissingTermManager {
   }
 
   private removeTermForcedToAppear(term: string) {
-    this.updateTermsForcedToAppear();
+    this.getUpdateTermsForcedToAppear();
     const termIndex = this.termForcedToAppear.indexOf(term);
     this.termForcedToAppear.splice(termIndex, 1);
-    this.queryStateModel.set('missingTerm', [...this.termForcedToAppear]);
+    this.setUpdateTermsForcedToAppear(this.termForcedToAppear);
     this.queryController.executeQuery();
   }
 
   private handleClearBreadcrumb() {
-    this.queryStateModel.set('missingTerm', []);
+    this.setUpdateTermsForcedToAppear([]);
+  }
+
+  private handleQueryChange(args: IAttributeChangedEventArg) {
+    this.getUpdateTermsForcedToAppear();
+    let termForcedToAppearCopy = [...this.termForcedToAppear];
+    this.termForcedToAppear.forEach(term => {
+      const regex = XRegExp(`${MissingTermManager.wordBoundary}(${term})${MissingTermManager.wordBoundary}`, 'g');
+      if (!regex.test(args.value)) {
+        const termIndex = termForcedToAppearCopy.indexOf(term);
+        termForcedToAppearCopy.splice(termIndex, 1);
+      }
+    });
+    this.setUpdateTermsForcedToAppear(termForcedToAppearCopy);
+    const breadcrumbSelector = document.querySelector('.CoveoBreadcrumb');
```

Comment:
You could use `this.searchInterface.getComponents('Breadcrumb')` instead, and check if the array is empty.

Reply:
This is not a component, so I don't have access to `this.searchInterface